src/accelerate/utils/modeling.py

```diff
@@ -991,10 +991,6 @@ def get_balanced_memory(
 
     if is_npu_available():
         expected_device_type = "npu"
-    elif is_mlu_available():
-        expected_device_type = "mlu"
```

Comment:
The issue you shared @faaany is about the `to() `methods for mlu. I'm not sure if `torch.device(d).type` will really return `cuda` with mlu. Do you have any insights @huismiling since you added the [support](https://github.com/huggingface/accelerate/pull/2552) to mlu ? To be safe, I would suggest reverting this change.

Reply:
good idea, I will update it. But I am also very curious about the behavior on npu.

Hi @statelesshz , could you help us verify what `torch.device(0)` returns on NPU? It is cuda or npu? Thanks a lot!

Reply:
@faaany  Hi, MLU devices type is `mlu` . 

```
>>> torch.device(0).type
'mlu'
```

Reply:
Thanks @huismiling! Could you update @faaany ?

Reply:
@huismiling thanks for updating!